File: Api/Dto/RecipeDto.cs
namespace Api.Dto
{
    public record RecipeDto(
        int Id,
        string Title,
        string Description,
        int CookingMinutes,
        int NumberOfServings,
        string Image,
        int AuthorId,
        List<TagDto> Tags,
        int CountOfLikes,
        int CountOfFavorites);
}

File: CullinaryBookSite/ClientApp/src/app/Dto/recipeDto.ts
import { TagDto } from "./tagDto";

export interface RecipeDto{
    id: number;
    title: string;
    description: string;
    cookingMinutes: number;
    numberOfServings: number;
    image: string;
    authorId: number;
    tags: TagDto[];
    countOfLikes: number;
    countOfFavorites: number;
}